### PR TITLE
ames-flow-kill

### PR DESCRIPTION
**Pull request by h5gq3**
close and delete state from ames flow after %leave task
web+urbitgraph://group/~middev/the-forge/graph/~littel-wolfur/writs-7082/170141184505118342639209431323694333952

### CODE REVIEW

File: pkg/arvo/sys/vane/ames.hoon [hunk 1]
```diff
@@ -1412,8 +1511,30 @@
         =/  sndr  [our our-life.channel]
         =/  rcvr  [ship her-life.channel]
         "plea {<sndr^rcvr^bone=bone^vane.plea^path.plea>}"
+    ::  since flow kill goes like:
+    ::  client vane cork task -> client ames pass cork as plea -> server ames sinks plea -> server ames +on-plea (we are here)
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:1515]: 80 characters

File: pkg/arvo/sys/vane/ames.hoon [hunk 2]
```diff
@@ -1412,8 +1511,30 @@
...
+    ?>  ?=([~ %known *] ship-state)
+    =/  =peer-state  +.u.ship-state
+    =/  =channel     [[our ship] now channel-state -.peer-state]
+    ::
+    =/  =plea  [%a /flow [%cork ~]]
+    ::
+    =^  =bone  ossuary.peer-state  (bind-duct ossuary.peer-state duct)
+    ::
+    =.  closing.peer-state  (~(put in closing.peer-state) bone)
+      abet:(on-memo:(make-peer-core peer-state channel) bone plea %plea)
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:1537]: de-indent this line

File: pkg/arvo/sys/vane/ames.hoon [hunk 3]
```diff
@@ -2150,10 +2273,22 @@
         ::  not a nack-trace bone; relay ack to client vane
         ::
         (emit (got-duct bone) %give %done error)
+      ::  +on-pump-cork: kill flow on cork sender side
+      ::
+      ++  on-pump-cork
+        ^+  peer-core
+        =.  by-duct.ossuary.peer-state  (~(del by by-duct.ossuary.peer-state) (got-duct bone))
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2280]: past 80 characters, I think

File: pkg/arvo/sys/vane/ames.hoon [hunk 4]
```diff
@@ -2150,10 +2273,22 @@
...
+        =.  by-bone.ossuary.peer-state  (~(del by by-bone.ossuary.peer-state) bone)
+        =.  snd.peer-state  (~(del by snd.peer-state) bone)
+        =.  rcv.peer-state  (~(del by rcv.peer-state) bone)
+        =.  corked.peer-state  (~(put in corked.peer-state) bone)
+        =.  closing.peer-state  (~(del in closing.peer-state) bone)
+          peer-core
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2286]: de-indent

File: pkg/arvo/sys/vane/ames.hoon [hunk 5]
```diff
@@ -2150,10 +2273,22 @@
...
       ::  +on-pump-send: emit message fragment requested by |message-pump
       ::
       ++  on-pump-send
-        |=(f=static-fragment (send-shut-packet bone [message-num %& +]:f))
+        |=  f=static-fragment
+          (send-shut-packet bone [message-num %& +]:f)
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2291]: de-indent

File: pkg/arvo/sys/vane/ames.hoon [hunk 6]
```diff
@@ -2195,8 +2331,25 @@
             ?-  -.gift
               %memo  (on-sink-memo [message-num message]:gift)
               %send  (on-sink-send [message-num ack-meat]:gift)
+              %cork  on-sink-cork
             ==
           $(sink-gifts t.sink-gifts)
+      ::  +on-sink-cork: handle flow kill after server ames has taken %done
+      ::
+      ++  on-sink-cork
+      :: resetting timer for boons
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2340]: multiple indentation problems in this arm

File: pkg/arvo/sys/vane/ames.hoon [hunk 7]
```diff
@@ -2230,10 +2383,11 @@
       ++  on-sink-boon
         |=  [=message-num message=*]
         ^+  peer-core
+        ?:  |((~(has in closing.peer-state) bone) (~(has in corked.peer-state) bone))  peer-core
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2386]: past 80 characters: this line, and the later conditionals that look similar

File: pkg/arvo/sys/vane/ames.hoon [hunk 8]
```diff
@@ -2292,6 +2447,9 @@
           ::  fresh plea; pass to client vane
           ::
           =+  ;;  =plea  message
+          ::  if this plea is %cork, put to closing
+          ::
+          =?  closing.peer-state  ?=([%cork *] payload.plea)  (~(put in closing.peer-state) bone)
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2452]: past 80

File: pkg/arvo/sys/vane/ames.hoon [hunk 9]
```diff
@@ -2312,7 +2470,7 @@
         ::
         (run-message-pump nack-trace-bone %memo message-blob)
       --
-    --
+  --
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2473]: indent

File: pkg/arvo/sys/vane/ames.hoon [hunk 10]
```diff
@@ -3138,7 +3297,9 @@
     =.  last-acked.state  +(last-acked.state)
     =?  nax.state  !ok  (~(put in nax.state) message-num)
     ::
-    =.  message-sink  (give %send message-num %| ok lag=`@dr`0)
+    =?  message-sink  cork  (give %send message-num %| ok %& lag=`@dr`0)
+    =?  message-sink  cork  (give %cork ~)
+    =?  message-sink  !cork  (give %send message-num %| ok %| lag=`@dr`0)
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:3302]: I think these three lines can be replaced by two:
```
=.  message-sink  (give %send message-num %| ok cork lag=`@dr`0)
=?  message-sink  cork  (give %cork ~)
```

File: pkg/arvo/sys/vane/gall.hoon [hunk 11]
```diff
@@ -488,6 +488,9 @@
       =/  stand
         (~(gut by outstanding.state) [wire hen] *(qeu remote-request))
       (~(put by outstanding.state) [wire hen] (~(put to stand) -.deal))
+    ?:  ?=(%leave -.deal)
+    =.  mo-core  (mo-pass wire note-arvo)
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/gall.hoon:492]: indent

**Review by belisarius222 — CHANGES_REQUESTED** [review]
Thanks for making this PR!

Initial review, just a bunch of style nitpicking.  Will review more thoroughly after cleanup.

**Comment by h5gq3** [comment]
fixed styling.
but the whole thing doesn't build. is this because this PR is not yet finished?

**Comment by joemfb** [comment]
This was included in #5644, which was included in #5886 and just released.

**Pull request closed without merging**